server/src/main/java/org/elasticsearch/rest/action/document/RestBulkAction.java

```diff
@@ -102,9 +103,11 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             boolean defaultRequireDataStream = request.paramAsBoolean(DocWriteRequest.REQUIRE_DATA_STREAM, false);
             bulkRequest.timeout(request.paramAsTime("timeout", BulkShardRequest.DEFAULT_TIMEOUT));
             bulkRequest.setRefreshPolicy(request.param("refresh"));
+            ReleasableBytesReference content = request.requiredReleasableContent();
+            content.mustIncRef();
```

Comment:
nit: I think the exception handling here is correct, but I wonder if we should use the same pattern as in `RestIndexAction` where we call `mustIncRef` within the lambda, just before `client.bulk`. That will be more robust to future changes that might introduce more exception paths around here.